apiVersion: serialmanipulator.timr.com/v1alpha1
metadata:
  name: robot
  description: Robot description
kind: SerialManipulator
spec:
  dhParams:
    theta: [0, 180, -90, 0, 0, 0]
    d: [0.1735, 0, 0, -0.1249, 0, 0.267]
    a: [0.027, 0.1629, 0.055, 0, 0, 0]
    alpha: [-90, 180, -90, 90, 90, 0]
    jointTypes: [0, 0, 0, 0, 0, 0]
  jointLimits:
    minJointPositions: [-180, 0, 0, -175, -120, -175]
    maxJointPositions: [180, 175, 150, 175, 120, 175]
    minJointVelocities: [-1, -2, -3, -4, -5, -6]
    maxJointVelocities: [1, 2, 3, 4, 5, 6]
    minJointAccelerations: [-1, -2, -3, -4, -5, -6]
    maxJointAccelerations: [1, 2, 3, 4, 5, 6]
  solverConfig:
    translationPriority: 1
    errorGain: 50
    jointDamping: 0.0001
    samplingTimeSec: 0.04
    isOpenLoop: true